### PR TITLE
Create TokenExchangeGrant model and add associations.

### PR DESCRIPTION
**Pull request by emilykim13**
## Description

This PR...

Adds TokenExchangeGrant model, factory, spec and migration files. Adds basic associations to TokenExchangeGrant model file. Tests associations in TokenExchangeGrant.

This is the a separate PR from the functional exchange changes since those changes will likely break several tests.

## Testing

Follow these steps to test this PR:

-  run `bundle exec rspec spec/models/oauth/token_exchange_grant_spec.rb`

### CODE REVIEW

File: db/migrate/20240722202054_create_oauth_token_exchange_grants.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+##
+# Creates the oauth_token_exchange_grants table.class CreateOAuthTokenExchangeGrants < ActiveRecord::Migration[7.0]
```
Comment (dickdavis) [inline db/migrate/20240722202054_create_oauth_token_exchange_grants.rb:4]: Looks like the comment and class definition somehow got joined together. You'll need to fix like this:
```
##
#  Creates the oauth_token_exchange_grants table
class CreateOAuthTokenExchangeGrants < ActiveRecord::Migration[7.0]
  def change
    ...
  end
end
```

File: app/models/oauth/token_exchange_grant.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+module OAuth
+  ##
+  # Models token exchange grant.
+  class TokenExchangeGrant < ApplicationRecord
+    include OAuth::SessionCreatable
+
+    belongs_to :user
+    belongs_to :oauth_client, class_name: 'OAuth::Client'
+    has_many :oauth_sessions,
```
Comment (dickdavis) [inline app/models/oauth/token_exchange_grant.rb:11]: ❓ Are you planning on adding the `grant` polymorphic column to the `oauth_sessions` table in this PR? If not, you'll want to take this out.

**Review by dickdavis — CHANGES_REQUESTED** [review]

File: db/migrate/20240722202054_create_oauth_token_exchange_grants.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+##
+#  Creates the oauth_token_exchange_grants table
+class CreateOAuthTokenExchangeGrants < ActiveRecord::Migration[7.0]
+  def change
+    table.references :user, null: false, foreign_key: true
```
Comment (dickdavis) [inline db/migrate/20240722202054_create_oauth_token_exchange_grants.rb:7]: Hmm, the `create_table` block somehow got removed. You'll need to wrap lines 7-10 in this block:

```ruby
create_table :oauth_token_exchange_grants, id: :uuid do |table|
  ...
end
```

**Review by dickdavis — CHANGES_REQUESTED** [review]

File: db/migrate/20240722202054_create_oauth_token_exchange_grants.rb (hunk 3 above)
Comment (emilykim13) [reply db/migrate/20240722202054_create_oauth_token_exchange_grants.rb]: Oops my bad! Added that back in

**Review by dickdavis — APPROVED** [review]
LGTM

**Pull request merged**